flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/options/JdbcOptions.java

```diff
@@ -67,11 +68,22 @@ public boolean equals(Object o) {
 				Objects.equals(driverName, options.driverName) &&
 				Objects.equals(username, options.username) &&
 				Objects.equals(password, options.password) &&
-				Objects.equals(dialect.getClass().getName(), options.dialect.getClass().getName());
+				Objects.equals(dialect.getClass().getName(), options.dialect.getClass().getName()) &&
+				Objects.equals(connectionCheckTimeoutSeconds, options.connectionCheckTimeoutSeconds);
 		} else {
 			return false;
 		}
 	}
+	@Override
+	public int hashCode() {
+		return Objects.hash(url,
+			tableName,
+			driverName,
+			username,
+			password,
+			dialect,
```

Comment:
Use `dialect.getClass().getName()` instead of `dialect`, otherwise, the `hashcode` is not compliant with `equals`.